types/validator_set.go

```diff
@@ -86,11 +86,13 @@ func (vals *ValidatorSet) IncrementProposerPriority(times int) {
 	// the 2nd check (threshold > 0) is merely a sanity check which could be
 	// removed if all tests would init. voting power appropriately;
 	// i.e. threshold should always be > 0
-	if diff > threshold && threshold > 0 {
+
+	for diff > threshold && threshold > 0 {
```

Comment:
This is guaranteed to terminate, right?

Reply:
I think so as the rescaling / normalisation should guarantee that after a finite (a few) number of iterations we reach diff <= thresold. I'd assume in the case of non-random prios, this should happen after one iter even almost always.